src/dagmc/tools/test_geom.cpp

```diff
@@ -0,0 +1,1158 @@
+                << tests[i].coords[1] << ", " << tests[i].coords[2]
+                << ").  Got " << names[result] << std::endl;
+      return MB_FAILURE;
+    }
+
+    // point_in_volume_slow doesn't to boundary.
+    if (tests[i].result == BOUNDARY)
+      continue;
+
+    rval = dagmc->point_in_volume_slow(vol, tests[i].coords, result);
+    CHKERR;
+
+    if (result != tests[i].result) {
+      std::cerr << "ERROR testing point_in_volume_slow[" << i << "]:" << std::endl
+                << "\tExpected " << names[tests[i].result]
+                << " for (" << tests[i].coords[0] << ", "
+                << tests[i].coords[1] << ", " << tests[i].coords[2]
+                << ").  Got " << names[result] << std::endl;
+      return MB_FAILURE;
+    }
+  }
+
+  return MB_SUCCESS;
+}
+
+ErrorCode overlap_test_tracking(DagMC* dagmc) {
+  /* Track a particle from left (-x) to right (+x) through an overlap.
+                ____________________
+                |                   |
+       -x <---  | region1 | overlap | region0 |   ---> +x
+                          |                   |
+                          ---------------------
+       -x <--- -1.0      0.0      0.01        1.0 ---> +x
+     surf_id:   10        4         8         2                     */
+
+  // get the surfaces and volumes
+  Tag dim_tag = dagmc->geom_tag();
+  Range surfs, explicit_vols;
+  const int two = 2, three = 3;
+  const void* ptrs[] = { &two, &three };
+  ErrorCode rval;
+  Interface* moab = dagmc->moab_instance();
+  rval = moab->get_entities_by_type_and_tag(0, MBENTITYSET, &dim_tag,
+                                            ptrs, 1, surfs);
+  CHKERR;
+  rval = moab->get_entities_by_type_and_tag(0, MBENTITYSET, &dim_tag,
+                                            ptrs + 1, 1, explicit_vols);
+  CHKERR;
+
+  if (explicit_vols.size() != 2) {
+    std::cerr << "ERROR: Expected 2 explicit volumes in input, found "
+              << explicit_vols.size() << std::endl;
+    return MB_FAILURE;
+  }
+  const unsigned num_surf = 12;
+  if (surfs.size() != num_surf) {
+    std::cerr << "ERROR: Expected " << num_surf << " surfaces in input, found "
+              << surfs.size() << std::endl;
+    return MB_FAILURE;
+  }
+  const EntityHandle explicit_vol = explicit_vols.front();
+
+  // start particle
+  double point[] = { -0.9, 0, 0 };
+  const double dir[] = { 1, 0, 0 };
+  EntityHandle vol = explicit_vol;
+  int result;
+  const int INSIDE = 1; // OUTSIDE = 0, BOUNDARY = -1;
+  rval = dagmc->point_in_volume(explicit_vol, point, result, dir);
+  CHKERR;
+  if (result != INSIDE) {
+    std::cerr << "ERROR: particle not inside explicit volume" << std::endl;
+    return MB_FAILURE;
+  }
+
+  // get next surface
+  double dist;
+  EntityHandle next_surf;
+  DagMC::RayHistory history;
+  rval = dagmc->ray_fire(vol, point, dir, next_surf, dist, &history);
+  CHKERR;
+  if (next_surf != surfs[7] || fabs(dist - 0.91) > 1e-6) {
```

Comment:
Readability: I would leave a comment explaining the value of `0.91`.

Reply:
@pshriwise @gonuke @makeclean any idea about that one ?

Reply:
This is basically a unit test, so that is the expected value of `dist`... perhaps
```suggestion
  expected_dist = 0.91
  if (next_surf != surfs[7] || fabs(dist - expected_dist) > 1e-6) {
```

Reply:
included in #676